benchmarks/distributed_contact/distributed_contact.cpp

```diff
@@ -0,0 +1,235 @@
+struct ExtractIndex
+{
+  template <typename Query, typename Value, typename Output>
+  KOKKOS_FUNCTION void operator()(Query const &, Value const &value,
+                                  Output const &out) const
+  {
+    out(value.index);
+  }
+};
+
+template <typename Coordinate>
+int main_(MPI_Comm const comm)
+{
+  ArborXBenchmark::TimeMonitor time_monitor;
+
+  using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+  using MemorySpace = typename ExecutionSpace::memory_space;
+
+  using Box = ArborX::Box<DIM, Coordinate>;
+
+  auto [points, triangles] = readTriangles<Coordinate>("unit_sphere.txt");
+  auto [primitives_v, predicates_v] = buildMesh(comm, points, triangles);
+
+  ExecutionSpace space;
+
+  Kokkos::View<Box *, MemorySpace> primitives(
+      Kokkos::view_alloc("Benchmark::primitives", Kokkos::WithoutInitializing),
+      primitives_v.size());
+  auto primitives_host = Kokkos::create_mirror_view(primitives);
+  for (size_t i = 0; i < primitives_v.size(); ++i)
+  {
+    Box box;
+    ArborX::Details::expand(box, primitives_v[i]);
+    primitives_host[i] = box;
+  }
+  Kokkos::deep_copy(space, primitives, primitives_host);
+
+  Kokkos::View<Box *, MemorySpace> predicates(
+      Kokkos::view_alloc("Benchmark::predicates", Kokkos::WithoutInitializing),
+      predicates_v.size());
+  auto predicates_host = Kokkos::create_mirror_view(predicates);
+  for (size_t i = 0; i < predicates_v.size(); ++i)
+  {
+    Box box;
+    ArborX::Details::expand(box, predicates_v[i]);
+    predicates_host[i] = box;
+  }
+  Kokkos::deep_copy(space, predicates, predicates_host);
+
+  auto construction_time = time_monitor.getNewTimer("construction");
+  MPI_Barrier(comm);
+  construction_time->start();
+  ArborX::DistributedTree<MemorySpace, ArborX::PairValueIndex<Box, int>> tree(
+      comm, space, ArborX::Experimental::attach_indices<int>(primitives));
+  construction_time->stop();
+
+  Kokkos::View<int *, MemorySpace> offsets("Benchmark::offsets", 0);
+  Kokkos::View<int *, MemorySpace> indices("Benchmark::indices", 0);
+  auto query_time = time_monitor.getNewTimer("query");
+  MPI_Barrier(comm);
+  query_time->start();
+  tree.query(space, ArborX::Experimental::make_intersects(predicates),
+             ExtractIndex{}, indices, offsets);
+  query_time->stop();
+
+  time_monitor.summarize(comm);
+
+  return 0;
+}
+
+int main(int argc, char *argv[])
+{
+  MPI_Init(&argc, &argv);
+
+  MPI_Comm const comm = MPI_COMM_WORLD;
+  int comm_rank;
+  MPI_Comm_rank(comm, &comm_rank);
+
+  if (comm_rank == 0)
+  {
+    std::cout << "ArborX version: " << ArborX::version() << std::endl;
+    std::cout << "ArborX hash   : " << ArborX::gitCommitHash() << std::endl;
+    std::cout << "Kokkos version: " << ArborX::Details::KokkosExt::version()
+              << std::endl;
+  }
+
+  // Strip "--help" and "--kokkos-help" from the flags passed to Kokkos if we
+  // are not on MPI rank 0 to prevent Kokkos from printing the help message
+  // multiply.
```

Comment:
```suggestion
  // multiple times.
```

Reply:
Copy-paste. It may be technically [correct](https://www.merriam-webster.com/dictionary/multiply), though unusual. We have a few comments using it this way, which should probably be changed too.